src/app/model/Pawn.ts

```diff
@@ -0,0 +1,73 @@
+import { Board } from './Board';
+import { Piece } from './Piece';
+import { PieceMovement } from './PieceMovement';
+import { Column, columns, Row, Side, Square } from './Types';
+
+const NEXT_ROW_DIFFERENCE = {
+  WHITE: 1,
+  BLACK: -1,
+};
+
+export class Pawn extends Piece implements PieceMovement {
+  private nextRowDifference = 1;
+
+  constructor(id: string, side: Side) {
+    super('uselessId', side);
+    if (this.side !== 'WHITE') this.nextRowDifference = -1;
+    this.nextRowDifference = this.side === 'WHITE' ? NEXT_ROW_DIFFERENCE.WHITE : NEXT_ROW_DIFFERENCE.BLACK;
+  }
```

Comment:
Chyba nie do końca się zrozumieliśmy. Chodziło mi o to, że możemy pozbyć się też IFa :) Pisząc o tak:
To też jest bardziej czytelnie po odnosisz się do konkretnej strony WHITE / BLACK, a nie mamy "jesli WHITE to to", to po co nam BLACK :) ? 

```suggestion
  constructor(id: string, side: Side) {
    super('uselessId', side);
    this.nextRowDifference = NEXT_ROW_DIFFERENCE[side]
  }
```

Ale fajna rzecz - obczaiłem jak działają sugestie zmian na GitHub w CodeReview :)

Reply:
ALE dodatków, jestem wielki przeciwnikiem umieszczania tego jako pole Pawn.
Pole klasy w jakiś sposób go opisuje, a dokładnie to opisuje instancje. Tutaj to ani nie powinno byc modyfikowane od utworzenia (bo pawn nie zmienia strony z białej na czarna itp.), a w takim razie można po prostu zamist wykorzystywać tą zmiennę odnosić się do `NEXT_ROW_DIFFERENCE[this.side]`
ALBO zrobić sobie metodę PRYWATNĄ która wykorzysta tą naszą stałą
```
type NextRowDifference = 1 | -1
const NEXT_ROW_DIFFERENCE: {[key: string]: NextRowDifference}= {
  WHITE: 1,
  BLACK: -1,
};

  private nextRowDifference(): NextRowDifference {
    return NEXT_ROW_DIFFERENCE[this.side];
  }
```
Ewentualnie ten dodatkowy typ nam mówi, że to jest nie jakikolwiek number a 1 lub -1. Im bardziej zawezamy typy tym oczywiscie lepiej :)